query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,484 @@
+    }
+
+    // prepare license if needed
+    if (metaAsObject.license) {
+      result.license = await prepareLicense(metaAsObject.license)
+    }
+
+    // prepare thumbnail photo asset if needed
+    if (metaAsObject.thumbnailPhoto !== undefined) {
+      const asset = await extractAsset({
+        assetIndex: metaAsObject.thumbnailPhoto,
+        assets: parameters.assets,
+        db: parameters.db,
+        blockNumber: parameters.blockNumber,
+        contentOwner: parameters.contentOwner,
+      })
+      integrateAsset('thumbnail', result, asset) // changes `result` inline!
+      delete metaAsObject.thumbnailPhoto
+    }
+
+    // prepare video asset if needed
+    if (metaAsObject.video !== undefined) {
+      const asset = await extractAsset({
+        assetIndex: metaAsObject.video,
+        assets: parameters.assets,
+        db: parameters.db,
+        blockNumber: parameters.blockNumber,
+        contentOwner: parameters.contentOwner,
+      })
+      integrateAsset('media', result, asset) // changes `result` inline!
+      delete metaAsObject.video
+    }
+
+    // prepare language if needed
+    if (metaAsObject.language) {
+      result.language = await prepareLanguage(metaAsObject.language, parameters.db, parameters.blockNumber)
+    }
+
+    // prepare information about media published somewhere else before Joystream if needed.
+    if (metaAsObject.publishedBeforeJoystream) {
+      // this will change the `channel`!
+      handlePublishedBeforeJoystream(result, metaAsObject.publishedBeforeJoystream.date)
+    }
+
+    return result as Partial<T>
+  }
+
+  // this should never happen
+  logger.error('Not implemented metadata type', {type})
+  throw `Not implemented metadata type`
+}
+
+export function convertContentActorToOwner(contentActor: ContentActor, channelId: BN): typeof DataObjectOwner {
+  const owner = new DataObjectOwnerChannel()
+  //owner.channel = channelId // TODO: make this work; it causes error `TypeError, message: attempted to use private field on non-instance`
+
+  return owner
+
+  /* contentActor is irrelevant now -> all video/channel content belongs to the channel
+  if (contentActor.isMember) {
+    const owner = new DataObjectOwnerMember()
+    owner.member = contentActor.asMember.toBn()
+
+    return owner
+  }
+
+  if (contentActor.isLead || contentActor.isCurator) {
+    const owner = new DataObjectOwnerChannel()
+    owner.channel = channelId
+
+    return owner
+  }
+
+  logger.error('Not implemented ContentActor type', {contentActor: contentActor.toString()})
+  throw 'Not-implemented ContentActor type used'
+  */
+}
+
+function handlePublishedBeforeJoystream(video: Partial<Video>, publishedAtString?: string) {
+  // published elsewhere before Joystream
+  if (publishedAtString) {
+    video.publishedBeforeJoystream = new Date(publishedAtString)
+  }
+
+  // unset publish info
+  video.publishedBeforeJoystream = undefined // plan deletion (will have effect when saved to db)
```

Comment:
did you miss a return statement in previous if block? , otherwise the value is always being set to undefined on the following line

Reply:
`return` was trully missing. Fixed